src/main/java/org/junit/rules/TemporaryFolder.java

```diff
@@ -216,10 +219,23 @@ public File newFolder() throws IOException {
     }
 
     private File createTemporaryFolderIn(File parentFolder) throws IOException {
-        File createdFolder = File.createTempFile("junit", "", parentFolder);
-        createdFolder.delete();
-        createdFolder.mkdir();
-        return createdFolder;
+        File createdFolder = null;
+        for (int i = 0; i < TEMP_DIR_ATTEMPTS; ++i) {
+            // Use createTempFile to get a suitable folder name.
+            String suffix = ".tmp";
+            File tmpFile = File.createTempFile(TMP_PREFIX, suffix, parentFolder);
+            tmpFile.delete();
```

Comment:
Please move this after creating a new temporary directory. It would protect us from race conditions  because no other JUnit thread or process could "steal" the path between the delete and the create (that is our theory as to the root cause of the bug).

Also, the code for `createTempFile()` did change.  See this blog post:

http://www.veracode.com/blog/2009/01/how-boring-flaws-become-interesting/